src/ast.ts

```diff
@@ -113,13 +113,13 @@ export namespace Expression {
     kind: 'InOp'
     lhs: Expression
     op: 'IN' | 'NOT IN'
-    subquery: Select
+    subquery: Select | Expression[]
```

Comment:
You should add a new type for the rhs of `IN`, something like

```
export type InSubquery = { kind: 'InSubquery', subquery: Select }
export type InExprList = { kind: 'InExprList', exprList: Expression[] }
export type InRhs = InSubquery | InExprList

export type InOp = {
  ...
  rhs: InRhs
}
```

This is because it's hard (or at least unreadable) to distinguish between a plain `Select` and `Expression[]`.

Reply:
And where you _need_ to distinguish between them is the NULL infering logic around `infer.ts:691`. According to the [docs](https://www.postgresql.org/docs/current/functions-comparisons.html#FUNCTIONS-COMPARISONS-IN-SCALAR), the result of the expression `lhs IN rhs` is `NULL` if
1. `lhs` is `NULL`
2. if `rhs` is a subquery that generates a `NULL` value and there's no match
3. if `rhs` is a list of values, one of which is `NULL` and there's no match

Looking at the code, only 1 is currently checked for, even though 2 should also be checked. This PR adds syntax support for 3, and would thus also need infering logic for 3.

Reply:
I've added support for (3) and a TODO to handle (2)